src/udx.c

```diff
@@ -705,15 +705,26 @@ process_packet (udx_socket_t *socket, char *buf, ssize_t buf_len, struct sockadd
   uint32_t seq = udx__swap_uint32_if_be(*(i++));
   uint32_t ack = udx__swap_uint32_if_be(*i);
 
-  buf += UDX_HEADER_SIZE;
-  buf_len -= UDX_HEADER_SIZE;
-
   udx_stream_t *stream = (udx_stream_t *) udx__cirbuf_get(socket->streams_by_id, local_id);
 
   if (stream == NULL || stream->status & UDX_STREAM_DEAD) return 0;
 
   // We expect this to be a stream packet from now on
 
+  if (stream->relay_to) {
+    udx_stream_t *relay = stream->relay_to;
+
+    if (relay->socket == NULL) return 0;
+
+    ((uint32_t *) buf)[1] = udx__swap_uint32_if_be(relay->remote_id);
+
+    uv_buf_t b = uv_buf_init(buf, buf_len);
+
+    udx__sendmsg(relay->socket, &b, 1, (struct sockaddr *) &relay->remote_addr, relay->remote_addr_len);
```

Comment:
Actually not sure if this would ever fail with `EAGAIN` as it's run at most once per poll, just after `udx__recvmsg()`.

Reply:
I __think__ it will, as the poll is only guaranteed to be ready for reads here, not writes.

Reply:
The packets are now copied to the send queue in case of `EAGAIN`.